tests/failpoints/cases/test_cmd_epoch_checker.rs

```diff
@@ -0,0 +1,373 @@
+    let (cb, cb_receivers) = make_cb(&write_req);
+    cluster
+        .sim
+        .rl()
+        .async_command_on_node(1, write_req, cb)
+        .unwrap();
+    cb_receivers.assert_err();
+
+    fail::remove(rollback_merge_fp);
+    sleep_ms(100);
+
+    // New write request can succeed.
+    let write_req = make_write_req(&mut cluster, b"a");
+    let (cb, cb_receivers) = make_cb(&write_req);
+    cluster
+        .sim
+        .rl()
+        .async_command_on_node(1, write_req, cb)
+        .unwrap();
+    cb_receivers.assert_ok();
+}
+
+#[test]
+fn test_reject_proposal_during_leader_transfer() {
+    let mut cluster = new_node_cluster(0, 2);
+    let pd_client = cluster.pd_client.clone();
+    pd_client.disable_default_operator();
+    let r = cluster.run_conf_change();
+    pd_client.must_add_peer(r, new_peer(2, 2));
+
+    // Don't allow leader transfer succeed if it is actually triggered.
+    cluster.add_send_filter(CloneFilterFactory(
+        RegionPacketFilter::new(1, 2)
+            .msg_type(MessageType::MsgTimeoutNow)
+            .direction(Direction::Recv),
+    ));
+
+    cluster.must_put(b"k", b"v");
+    cluster.transfer_leader(r, new_peer(2, 2));
+    // The leader can't change to transferring state immediately due to pre-transfer-leader
+    // feature, so wait for a while.
+    sleep_ms(100);
+    assert_ne!(cluster.leader_of_region(r).unwrap(), new_peer(2, 2));
+
+    let write_req = make_write_req(&mut cluster, b"k");
+    let (cb, cb_receivers) = make_cb(&write_req);
+    cluster
+        .sim
+        .rl()
+        .async_command_on_node(1, write_req, cb)
+        .unwrap();
+    cb_receivers.assert_err();
+    cluster.clear_send_filters();
+}
+
+#[test]
+fn test_accept_proposal_during_conf_change() {
+    let mut cluster = new_node_cluster(0, 2);
+    cluster.pd_client.disable_default_operator();
+    let r = cluster.run_conf_change();
+    cluster.must_put(b"a", b"v");
+
+    let conf_change_fp = "apply_on_conf_change_all_1";
+    fail::cfg(conf_change_fp, "pause").unwrap();
+    let add_peer_rx = cluster.async_add_peer(r, new_peer(2, 2)).unwrap();
+    add_peer_rx
+        .recv_timeout(Duration::from_millis(100))
+        .unwrap_err();
+
+    // Conf change doesn't affect proposals.
+    let write_req = make_write_req(&mut cluster, b"k");
+    let (cb, cb_receivers) = make_cb(&write_req);
+    cluster
+        .sim
+        .rl()
+        .async_command_on_node(1, write_req, cb)
+        .unwrap();
+    cb_receivers
+        .committed
+        .recv_timeout(Duration::from_millis(100))
+        .unwrap();
+    cb_receivers.proposed.try_recv().unwrap();
+
+    fail::remove(conf_change_fp);
+    assert!(!add_peer_rx
+        .recv_timeout(Duration::from_millis(100))
+        .unwrap()
+        .get_header()
+        .has_error());
+    assert!(!cb_receivers
+        .applied
+        .recv_timeout(Duration::from_millis(100))
+        .unwrap()
+        .get_header()
+        .has_error(),);
```

Comment:
These two time are too short. Maybe 1 second is better.
And the `,` should be removed.